plugins/incremental-ingestion-backend/api-report.md

```diff
@@ -0,0 +1,261 @@
+## API Report File for "@backstage/plugin-incremental-ingestion-backend"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+/// <reference types="express" />
+
+import { CatalogBuilder } from '@backstage/plugin-catalog-backend';
+import type { Config } from '@backstage/config';
+import type { DeferredEntity } from '@backstage/plugin-catalog-backend';
+import { Duration } from 'luxon';
+import type { DurationObjectUnits } from 'luxon';
+import type { EntityProviderConnection } from '@backstage/plugin-catalog-backend';
+import { Knex } from 'knex';
+import type { Logger } from 'winston';
+import type { PermissionAuthorizer } from '@backstage/plugin-permission-common';
+import type { PluginDatabaseManager } from '@backstage/backend-common';
+import type { PluginTaskScheduler } from '@backstage/backend-tasks';
+import { Router } from 'express';
+import type { TaskFunction } from '@backstage/backend-tasks';
+import type { UrlReader } from '@backstage/backend-common';
+
+// @public
+export interface EntityIteratorResult<T> {
+  cursor: T;
+  done: boolean;
+  entities: DeferredEntity[];
+}
```

Comment:
How about just making `entities` and `cursor` optional? It accomplishes the same thing, without the complexity of a new union type.

Reply:
I've made both optional. Seems to solve the problem.

Reply:
Well, that's a weaker type, since then you'll be able to return `{ done: false }` - what should the framework do then? There are no entities to consume, and no cursor to advance with. Discriminated unions are fine, we use them a lot and it's pretty common in typescript since it's so powerful.

Reply:
Hmmm. That's a good point. Since that should never happen, perhaps it should trigger an error state?

Reply:
Where though? In the caller?

Nah I say, just bite the bullet and make the type the correct one as suggested to begin with.

Reply:
You're right, that's a better solution. Done!